src/hotspot/share/gc/g1/g1ParScanThreadState.cpp

```diff
@@ -363,7 +358,9 @@ G1HeapRegionAttr G1ParScanThreadState::next_region_attr(G1HeapRegionAttr const r
       return region_attr;
     }
   }
-  return dest(region_attr);
+  assert(region_attr.is_young() || region_attr.is_old(), "must be either Young or Old");
```

Comment:
I think I'd rather have this assert at the beginning of the function.

Reply:
Revised.